### ISSUE TITLE
Feature request: Checkpoint on time delta

### ISSUE THREAD

**Issue by ChrisCummins** [jcjohnson/torch-rnn#125]
It would be nice if there was the ability to create checkpoints based on the amount of time spent training, rather than a fixed number of epochs, e.g. `-checkpoint_every 30m` or `-checkpoint_every 2h`.

Also a `-max_checkpoints` flag would be nice so that you can have a rotating, finite number of checkpoint files. I have a long running job that's created over 125GB of checkpoints and is hitting the HDD capacity cap.

**Comment by dgcrouse** [comment]
I agree. I am also working on specifying checkpoints by epoch rather than batch, e.g. so ```-checkpoint_every 0.5e``` does a checkpoint every half epoch.

**Comment by dgcrouse** [comment]
@ChrisCummins any progress on implementing this?
